File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var cur = prices[0];
    var profit = 0;
    
    for(var i = 0; i < prices.length; i++){
        if(prices[i] < cur){
            cur = prices[i];
        }
        else{
            let temp = Math.max(prices[i]-cur,profit);
            profit = temp;
        }
    }
    return profit
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        res = 0
        big = -1
        
        for i in range(len(prices) - 1, -1, -1):
            if prices[i] > big:
                big = prices[i]
            
            res = max(big - prices[i], res)
        
        return res